src/sfloader/fluid_defsfont.c

```diff
@@ -1562,27 +1578,32 @@ fluid_zone_mod_source_import_sfont(unsigned char *src, unsigned char *flags, uns
     return TRUE;
 }
 
-/*
+/**
  * fluid_zone_mod_import_sfont
  * Imports modulators from sfzone to modulators list mod.
- * @param zone_name, zone name.
- * @param mod, address of pointer on modulators list to return.
- * @param sfzone, pointer on soundfont zone.
+ * @param mod -  address of pointer on modulators list to return.
+ * @param sfmod - pointer on the modulator list.
  * @return FLUID_OK if success, FLUID_FAILED otherwise.
  */
 static int
-fluid_zone_mod_import_sfont(char *zone_name, fluid_mod_t **mod, SFZone *sfzone)
+fluid_mod_import_sfont(fluid_mod_t **mod, fluid_list_t *sfmod)
 {
     fluid_list_t *r;
     int count;
 
     /* Import the modulators (only SF2.1 and higher) */
-    for(count = 0, r = sfzone->mod; r != NULL; count++)
+    for(count = 0, r = sfmod; r != NULL; count++)
     {
 
-        SFMod *mod_src = (SFMod *)fluid_list_get(r);
+        SFMod *mod_src = (SFMod*)(r->data);
         fluid_mod_t *mod_dest = new_fluid_mod();
```

Comment:
The modulator allocated in this line and later appended to the `defsfont->default_mod_list` is never freed. You need to call `delete_fluid_list_mod(defsfont->default_mod_list)` in `delete_fluid_defsfont()`.

Reply:
I think you've missed this one.

Reply:
I think you've missed this one